modules/apps/message-boards/message-boards-comment/src/main/java/com/liferay/message/boards/comment/internal/MBCommentManagerImpl.java

```diff
@@ -319,10 +318,8 @@ public Discussion getDiscussion(
 	}
 
 	@Override
-	public DiscussionPermission getDiscussionPermission(
-		PermissionChecker permissionChecker) {
-
-		return new MBDiscussionPermissionImpl(permissionChecker);
+	public DiscussionPermission getDiscussionPermission() {
+		return new MBDiscussionPermissionImpl();
```

Comment:
@W2jx @yuhai I wonder if this is really necessary. You first change it to a class without constructor, and then change it into a component. Why not directly make it a component and apply?